### PR TITLE
Fix issue 5317, when asking /api/checkouts/{id} for the available pay…

### LINKED ISSUE #5317: [ApiBundle] GET on /api/checkouts/{id} raises a 500
{
  "code": 500,
  "message": "The option \"criteria\" does not exist. Defined options are: \"action\", \"allow_extra_fields\", \"attr\", \"auto_initialize\", \"block_name\", \"by_reference\", \"cascade_validation\", \"compound\", \"constraints\", \"csrf_field_name\", \"csrf_message\", \"csrf_protection\", \"csrf_provider\", \"csrf_token_id\", \"csrf_token_manager\", \"data\", \"data_class\", \"description\", \"disabled\", \"empty_data\", \"error_bubbling\", \"error_mapping\", \"extra_fields_message\", \"inherit_data\", \"intention\", \"invalid_message\", \"invalid_message_parameters\", \"label\", \"label_attr\", \"label_format\", \"mapped\", \"max_length\", \"method\", \"pattern\", \"post_max_size_message\", \"property_path\", \"read_only\", \"required\", \"translation_domain\", \"trim\", \"validation_groups\", \"virtual\"."
}

Problems comes from ShippingStepType.php.

Working on PR right now

### PR DESCRIPTION
**Pull request by ylastapis**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | fixes #5317 
| License         | MIT

### CODE REVIEW

**Comment by ylastapis** [comment]
cleanup of https://github.com/Sylius/Sylius/pull/5345

File: docs/api/checkouts.rst [hunk 1]
```diff
@@ -785,7 +770,262 @@ Final response contains the full order information, now you can call the purchas
...
+                "locked": false,
+                "neutral": false,
+                "updated_at": "2014-12-15T14:30:41+0000"
+            }
+        ],
+        "adjustments_total": 4750,
+        "billing_address": {
+            "_links": {
+                "country": {
+                    "href": "/app_dev.php/api/countries/9"
```
Comment (lchrusciel) [inline docs/api/checkouts.rst]: `app_dev.php` part should be skipped in documentation, I think.

File: docs/api/checkouts.rst [hunk 2]
```diff
@@ -785,7 +770,262 @@ Final response contains the full order information, now you can call the purchas
...
+            "city": "New York",
+            "created_at": "2014-12-15T13:37:28+0000",
+            "first_name": "John",
+            "id": 105,
+            "last_name": "Doe",
+            "postcode": "12435",
+            "street": "Test",
+            "updated_at": "2014-12-15T13:37:29+0000"
+        },
+        "total": 1504750
```
Comment (lchrusciel) [inline docs/api/checkouts.rst]: missing comma at the end of line

File: docs/api/checkouts.rst [hunk 3]
```diff
@@ -785,7 +770,262 @@ Final response contains the full order information, now you can call the purchas
...
+
```
Comment (lchrusciel) [inline docs/api/checkouts.rst]: redundant

File: docs/api/checkouts.rst (hunk 1 above)
Comment (ylastapis) [inline docs/api/checkouts.rst]: @lchrusciel No opinion about that, I can push a new version by replacing all

File: docs/api/checkouts.rst (hunk 2 above)
Comment (ylastapis) [inline docs/api/checkouts.rst]: Missed that one, well done :)

File: docs/api/checkouts.rst (hunk 3 above)
Comment (ylastapis) [inline docs/api/checkouts.rst]: I don't understand what you want me to do on that particular point ;)

File: docs/api/checkouts.rst [hunk 4]
```diff
@@ -785,7 +770,262 @@ Final response contains the full order information, now you can call the purchas
...
+                "locked": false,
+                "neutral": false,
+                "updated_at": "2014-12-15T14:30:41+0000"
+            },
+            {
+                "amount": -250,
+                "created_at": "2014-12-15T14:30:41+0000",
+                "description": "Christmas Sale for orders over 100 EUR.",
+                "id": 208,
+                "type": "promotion",
```
Comment (Zales0123) [inline docs/api/checkouts.rst]: In fact, there is no ``promotion`` adjustment type. You can choose between ``order_item_promotion``, ``order_promotion``, ``order_shipping_promotion`` and ``order_unit_promotion`` ;)

File: docs/api/checkouts.rst [hunk 5]
```diff
@@ -785,7 +770,262 @@ Final response contains the full order information, now you can call the purchas
...
+                }
+            },
+            "city": "New York",
+            "created_at": "2014-12-15T13:37:28+0000",
+            "first_name": "John",
+            "id": 106,
+            "last_name": "Doe",
+            "postcode": "12435",
+            "street": "Test",
+            "updated_at": "2014-12-15T13:37:29+0000"
```
Comment (Zales0123) [inline docs/api/checkouts.rst]: I would keep the same order as in model:
1. id
2. firstName
3. lastName
4. where is the *countryCode*?
5. street
6. city
7. postcode
and then
8. createdAt
9. updatedAt

Comment (Zales0123) [inline docs/api/checkouts.rst]: Where is the ``country_code``?

Comment (ylastapis) [inline docs/api/checkouts.rst]: It seems it has been replaced by a HATEOAS link on latest Sylius updates :)

File: docs/api/checkouts.rst (hunk 4 above)
Comment (ylastapis) [inline docs/api/checkouts.rst]: "No more" promotion you mean :)
Doc hasn't been merged and it's already obsolete ^^

**Comment by ylastapis** [comment]
Anyway the point was to fix an annoying issue in the checkout process through API :)

**Comment by pjedrzejewski** [comment]
Thank you Yann! :)

**Pull request merged**